File: Basic/Test_N-thTribonacciNumber.py
# https://leetcode.com/problems/n-th-tribonacci-number/

class Solution:
    def tribonacci(self, n: int) -> int:
        Dynamic = [0, 1, 1]
        for i in range(n + 1):
            if n <= len(Dynamic) - 1:
                return Dynamic[n]
            Dynamic.append(Dynamic[i]+Dynamic[i + 1]+Dynamic[i+2])


s = Solution()
print(s.tribonacci(4))
for i in range(10):
    print(s.tribonacci(i))

File: Basic/Test_N-thTribonacciNumber.c
#include <stdio.h>

int tribonacci(int n)
{
    int dynamic[38] = {0};
    for (int i = 0; i < 3; i++)
    {
        if (i != 0)
        {
            dynamic[i] = 1;
        }
    }
    int actualSize = 3;
    for (int i = 0; i < n + 1; i++)
    {
        if (n <= actualSize - 1)
        {
            return dynamic[n];
        }
        dynamic[i + 3] = dynamic[i] + dynamic[i + 1] + dynamic[i + 2];
        actualSize += 1;
    }
}

int main()
{
    printf("%d ", tribonacci(36));
    return 0;
}